### ISSUE TITLE
PackageDeploymentManager.IsPackageReadyOrNewerAvailable() throws NotImplementedException even if IsPackageDeploymentFeatureSupported

### ISSUE THREAD

**Issue by whiskhub** [microsoft/WindowsAppSDK#4817]
### Describe the bug

`PackageDeploymentManager.IsPackageReadyOrNewerAvailable()` throws `NotImplementedException` even though `PackageDeploymentManager.IsPackageDeploymentFeatureSupported(PackageDeploymentFeature.IsPackageReadyOrNewerAvailable)` returns true.

Tested on Windows 11 24H2 (26100).

Note: it would be nice if on the docs page https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.windows.management.deployment.packagedeploymentmanager.ispackagereadyorneweravailable the requirement to call `PackageDeploymentManager.IsPackageDeploymentFeatureSupported` would be documented, together with a minimum supported OS version where this is expected to work.

### Steps to reproduce the bug

```
var pdm = Microsoft.Windows.Management.Deployment.PackageDeploymentManager.GetDefault();
var isSupported = PackageDeploymentManager.IsPackageDeploymentFeatureSupported(PackageDeploymentFeature.IsPackageReadyOrNewerAvailable);
if (isSupported) 
{ 
    var isNewerAvailable = pdm.IsPackageReadyOrNewerAvailable(Package.Current.Id.FullName); // throws System.NotImplementedException
}
```

### Expected behavior

Method does not throw and API works as expected.

### NuGet package version

Windows App SDK 1.6.1: 1.6.240923002

### Packaging type

Packaged (MSIX)

### IDE

Visual Studio 2022

**Comment by mominshaikhdevs** [comment]
@DrusTheAxe @manodasanW

**Comment by tonymet** [comment]
@DrusTheAxe  was this one fixed for version 1.6.241114003 ?  I'm seeing the exception with `ResetPackageAsync` https://github.com/microsoft/WindowsAppSDK/issues/5031
